### PR TITLE
Feature/linkify username

### PR DESCRIPTION
**Pull request by nadirabbas**
Putting a URL in this field makes usernames clickable.

image
image

### CODE REVIEW

File: kousa/priv/repo/migrations/20210221014438_external_profile_link.exs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+defmodule Beef.Repo.Migrations.ExternalProfileLink do
+  use Ecto.Migration
+
+  def change do
+    alter table(:users) do
+      add :externalProfileLink, :text, default: nil, allow_nil: true
```
Comment (mohammedgad) [inline kousa/priv/repo/migrations/20210221014438_external_profile_link.exs]: that should be a string.

File: kousa/lib/beef/user.ex [hunk 2]
```diff
@@ -48,18 +50,19 @@ defmodule Beef.User do
...
 
   def edit_changeset(user, attrs) do
     user
-    |> cast(attrs, [:id, :username, :bio, :displayName])
+    |> cast(attrs, [:id, :username, :bio, :displayName, :externalProfileLink])
     |> validate_required([:username, :bio, :displayName])
     |> validate_length(:bio, min: 2, max: 160)
     |> validate_length(:displayName, min: 2, max: 50)
     |> validate_format(:username, ~r/^(\w){4,15}$/)
+    |> validate_format(:externalProfileLink, ~r/^(http|https):\/\/[^ "]{1,255}$/)
```
Comment (mohammedgad) [inline kousa/lib/beef/user.ex]: hi @nadirabbas I think the URL edit can't be nil because the validation `{1,255}` can you try to make it `{0,255}`"?

**Review by mohammedgad — CHANGES_REQUESTED** [review]

File: kousa/lib/beef/user.ex (hunk 2 above)
Comment (nadirabbas) [reply kousa/lib/beef/user.ex]: oh damn

Comment (nadirabbas) [reply kousa/lib/beef/user.ex]: It works @mohammedgad. Well, that was a dumb mistake. Nice catch 🎉

**Comment by benawad** [comment]
I don't want a separate link for username atm. I only want to linkify bio

**Comment by nadirabbas** [comment]
@benawad I guess I misunderstood yesterday :/

**Comment by nadirabbas** [comment]
I Will do that instead, sorry I misunderstood

**Comment by benawad** [comment]
it's all good, I could have been clearer

**Pull request closed without merging**